src/EndpointV2/EndpointV2Middleware.php

```diff
@@ -303,4 +311,50 @@ private function normalizeAuthScheme(array $authScheme) : array
 
         return $normalizedAuthScheme;
     }
+
+    /**
+     * This method tries to resolve an AccountId parameter from a resolved identity.
+     * We will just perform this operation if the parameter AccountId is part of the ruleset parameters and
+     * AccountIdEndpointMode is not disabled, otherwise, we will ignore it.
+     * We also are going to resolve identity from a supplied credentials provider,
+     * and we will inject the resolved identity into the $command['@context'] property, so it can be reused along the way.
+     *
+     * @param array $commandArgs
+     * @param array $rulesetParams
+     * @return void
+     */
+    private function resolveAccountIdArgument(array &$commandArgs, array $rulesetParams)
+    {
+        static $ACCOUNT_ID_PARAM = 'AccountId';
+        if (!in_array($ACCOUNT_ID_PARAM, array_keys($rulesetParams))) {
+            return;
+        }
+
+        $accountIdEndpointMode = $this->clientArgs['AccountIdEndpointMode'];
+        if ($accountIdEndpointMode === 'disabled') {
+            return;
+        }
+
+        $credentialsProviderFn = $this->credentialsProvider;
+        $identity = $credentialsProviderFn()->wait();
```

Comment:
I'd do a check here to see if the identity is set on the command already.  It might be, dependent on whether or not a service models auth schemes using the `auth` trait.